zenoh-jni/src/utils.rs

```diff
@@ -49,6 +50,19 @@ pub(crate) fn get_callback_global_ref(
     })
 }
 
+/// Helper function to convert a JByteArray into a Vec<u8>.
+pub(crate) fn decode_byte_array(env: &JNIEnv<'_>, payload: JByteArray) -> Result<Vec<u8>> {
+    let payload_len = env
+        .get_array_length(&payload)
+        .map(|length| length as usize)
+        .map_err(|err| Error::Jni(err.to_string()))?;
+    let mut buff = vec![0; payload_len];
+    env.get_byte_array_region(payload, 0, &mut buff[..])
+        .map_err(|err| Error::Jni(err.to_string()))?;
+    let buff: Vec<u8> = buff.iter().map(|&x| x as u8).collect();
```

Comment:
This implies a full copy (and additional allocation) just to transform `Vec<i8>` into `Vec<u8>`. A transmute here would be perfectly safe and much less costly.

Reply:
Oh nice, I didn't remember that resource.